### DISCUSSION TITLE
Improvement of LoadAndPlay method of the MidiPlayer.cs

### DISCUSSION THREAD

**Discussion by melanchall** [QL-Win/QuickLook#1520] [General]
Hi,

I'm the author of the DryWetMIDI library. First of all, thank you for using it!

Just want to suggest a small improvement in the code of the `LoadAndPlay` method within the `MidiPlayer.cs` file. You have such lines:

```csharp
// Playback supported by DryWetMidi will block the current thread
// So we should run it in a new thread
_ = Task.Run(() => _playback?.Play());
```

But playback can run in two modes: blocking and non-blocking, as stated in [the docs](https://melanchall.github.io/drywetmidi/articles/playback/Overview.html#non-blocking-playback). So instead of the lines above just write:

```csharp
_playback?.Start();
```

Thank you again and happy new year! :tada:

**Comment by emako** [comment]
Good and feasible!

**Comment by emako** [comment]
Done e70c23a
